pkg/codegen/internal/test/testdata/output-funcs/py_tests/getIntegrationRuntimeObjectMetadatum.py

```diff
@@ -0,0 +1,96 @@
+class GetIntegrationRuntimeObjectMetadatumResult:
+    """
+    A list of SSIS object metadata.
+    """
+    def __init__(__self__, next_link=None, value=None):
+        if next_link and not isinstance(next_link, str):
+            raise TypeError("Expected argument 'next_link' to be a str")
+        pulumi.set(__self__, "next_link", next_link)
+        if value and not isinstance(value, list):
+            raise TypeError("Expected argument 'value' to be a list")
+        pulumi.set(__self__, "value", value)
+
+    @property
+    @pulumi.getter(name="nextLink")
+    def next_link(self) -> Optional[str]:
+        """
+        The link to the next page of results, if any remaining results exist.
+        """
+        return pulumi.get(self, "next_link")
+
+    @property
+    @pulumi.getter
+    def value(self) -> Optional[Sequence[Any]]:
+        """
+        List of SSIS object metadata.
+        """
+        return pulumi.get(self, "value")
+
+
+class AwaitableGetIntegrationRuntimeObjectMetadatumResult(GetIntegrationRuntimeObjectMetadatumResult):
+    # pylint: disable=using-constant-test
+    def __await__(self):
+        if False:
+            yield self
+        return GetIntegrationRuntimeObjectMetadatumResult(
+            next_link=self.next_link,
+            value=self.value)
+
+
+def get_integration_runtime_object_metadatum(factory_name: Optional[str] = None,
+                                             integration_runtime_name: Optional[str] = None,
+                                             metadata_path: Optional[str] = None,
+                                             resource_group_name: Optional[str] = None,
+                                             opts: Optional[pulumi.InvokeOptions] = None) -> AwaitableGetIntegrationRuntimeObjectMetadatumResult:
+    """
+    A list of SSIS object metadata.
+    API Version: 2018-06-01.
+
+
+    :param str factory_name: The factory name.
+    :param str integration_runtime_name: The integration runtime name.
+    :param str metadata_path: Metadata path.
+    :param str resource_group_name: The resource group name.
+    """
+    __args__ = dict()
+    __args__['factoryName'] = factory_name
+    __args__['integrationRuntimeName'] = integration_runtime_name
+    __args__['metadataPath'] = metadata_path
+    __args__['resourceGroupName'] = resource_group_name
+    if opts is None:
+        opts = pulumi.InvokeOptions()
+    if opts.version is None:
+        opts.version = _utilities.get_version()
+    __ret__ = pulumi.runtime.invoke('azure-native:codegentest:getIntegrationRuntimeObjectMetadatum', __args__, opts=opts, typ=GetIntegrationRuntimeObjectMetadatumResult).value
+
+    return AwaitableGetIntegrationRuntimeObjectMetadatumResult(
+        next_link=__ret__.next_link,
+        value=__ret__.value)
+
+
+@_utilities.lift_output_func(get_integration_runtime_object_metadatum)
+def get_integration_runtime_object_metadatum_output(factory_name: Optional[pulumi.Input[str]] = None,
```

Comment:
Is the plan to have the args of the `_output` variant follow the exact same pattern as the source function (where all args are optional with `= None`, w/ pending runtime check to ensure required args are set) for consistency, or do we want to use the "improved" signature uses by methods and args classse, where we force args be specified by keyword (via a `*` arg, and sort required args first and annotate as non-optional, followed by optional args set to `= None`)? The former (what you have currently in this PR) sounds fine to me, but I was just wondering.

Reply:
I see it's the same comment you had in https://github.com/pulumi/pulumi/pull/7825

I think we should opt for consistency here and consider switching to the improved signature in both versions of the function later, which would be a breaking change for consumers calling arguments positionally, right? 

I started trying to jot down an issue but I think it's best if you do it, I have some doubt if it is unconditionally a good idea. The '*' trick to require call sites to use keywords is a good idea in my book, in fact I'd require all methods to use keywords if it was up to me (like in Smalltalk) but what I have doubts about is sorting the required arguments. 

Sorting.. seems weird to me. Either the schema has order, in which case the schema authors can arrange the most aesthetically pleasing order of these positional arguments, or it does not have order. In case it does not have order, sorting makes arbitrary decisions that can feel ugly. If all args required keywords the caller could make the decision? 

Checking up how it works, perhaps this:

```
def foo_bar(*, a, b=None, c=None):
    pass

# foo_bar()
# TypeError: foo_bar() missing 1 required keyword-only argument: 'a

foo_bar(a=1)
```

Also thinking out loud, we might need the order in the schema at least for deciding in which order we document parameters. We could have data on which parameters are most frequently set to make them float up in the docs.

Reply:
Also let me actually test if positionally passing arguments even works right now through the `lift_output_func` combinator.. I don't think I was careful enough to test this...

Reply:
That didn't work in fact.. added some tests now and fixed the decorator to use `inspect` to bind correctly.